src/types/any.rs

```diff
@@ -2184,6 +2184,7 @@ impl<'py> PyAnyMethods<'py> for Bound<'py, PyAny> {
 }
 
 #[cfg(test)]
+#[cfg_attr(not(feature = "gil-refs"), allow(deprecated))]
```

Comment:
I felt like the most practical option w.r.t keeping coverage up and test churn down during the initial implementation phase was to just add this conditional allow to tests at quite a wide level for now.

Then on the way from 0.21 to 0.22 when we're adding the full feature gate we can tidy up tests and make these allows narrower.